recipes/python/backup-restore/oracle_db_clone.py

```diff
@@ -0,0 +1,87 @@
+
+    try:
+        # run asset-service to get the asset id
+        RECOVERY_ASSET_ID = workload_oracle.get_oracle_asset_info(BASEURL, TOKEN, ASSET_TYPE, ARGS.source_oracle_db)
+        print(f"Oracle Asset Id:[{RECOVERY_ASSET_ID}]")
+        # run recovery-point-service to get recovery-point-id
+        RECOVERY_POINT_ID = common.get_recovery_points(BASEURL, TOKEN, WORKLOAD_TYPE, RECOVERY_ASSET_ID)
+        print(f"Oracle Recovery Point Id:[{RECOVERY_POINT_ID}]")
+
+        if (ARGS.pdb_clone):
+            preCheckURL = f"{BASEURL}recovery/workloads/oracle/scenarios/pdb-complete-clone/pre-recovery-check"
+            cloneURL = f"{BASEURL}recovery/workloads/oracle/scenarios/pdb-complete-clone/recover"
+            recovery_input_json = "post_oracle_pdb_clone.json"
+            data = workload_oracle.create_pdb_clone_request_payload(recovery_input_json, RECOVERY_POINT_ID, ARGS.credential_id, ARGS.target_oracle_server,
+                ARGS.clone_db_file_path, ARGS.oracle_home, ARGS.oracle_base_config, ARGS.avail_oracle_sid, ARGS.clone_aux_file_path)
+        else:
+            preCheckURL = f"{BASEURL}recovery/workloads/oracle/scenarios/database-complete-clone/pre-recovery-check"
+            cloneURL = f"{BASEURL}recovery/workloads/oracle/scenarios/database-complete-clone/recover"
+            recovery_input_json = "post_oracle_cdb_clone.json"
+            data = workload_oracle.create_cdb_clone_request_payload(recovery_input_json, RECOVERY_POINT_ID, ARGS.os_oracle_domain, ARGS.os_oracle_username,
+                ARGS.os_oracle_password, ARGS.target_oracle_server, ARGS.clone_db_file_path, ARGS.oracle_home, ARGS.oracle_base_config)
+        
+        # run a pre-recovery-check
+        print("Perform a pre-recovery-check")
+        status_code, response_text = workload_oracle.create_oracle_recovery_request(preCheckURL, TOKEN, data)
+
+        common.validate_response(status_code, 200, response_text)
+        print(f"Pre-recovery-check response :{response_text['data']}")
+        preCheckFail = 0
+        for _result in response_text['data']:
+            if (_result['attributes']['result'] != 'Passed'):
+                print(f"FAIL REASON: {_result['attributes']['description']}")
+                preCheckFail += 1
+
+        if ((not preCheckFail) or (ARGS.force_clone)):
+            # run a clone
+            print("Perform a database clone")
+
+            status_code, response_text = workload_oracle.create_oracle_recovery_request(cloneURL, TOKEN, data)
+            
+            common.validate_response(status_code, 201, response_text)
+            CLONE_JOBID = response_text['data']['id']
+            print(f"Clone initiated , follow Job # :[{CLONE_JOBID}]")
+        else:
+            print("Pre-recovery check failed. Force kick off clone using --force_clone")
+    
+    finally:
+        print("To cleanup the cloned instance, run dbca. Add the instance to /etc/oratab to be discovered by oracle for cleanup.")
```

Comment:
no, for windows they only need to run dbca. so thats the first half and the second part in case the instance is not discovered and doesn't appear on dbca